openedx/features/subscriptions/api/v1/views.py

```diff
@@ -32,7 +32,7 @@ def get_queryset(self):
         if return_only_valid_subscriptions:
             return UserSubscription.get_valid_subscriptions(self.request.site, username=username)
 
-        return UserSubscription.objects.filter(user_username=username)
+        return UserSubscription.objects.filter(user__username=username)
```

Comment:
@rehan99000  How come it was working before?

Reply:
All user subscriptions are not being fetched anywhere. since valid subscriptions are returned before this point, this got missed.